### PR TITLE
DOC: Updated aggregate docstring

### PR DESCRIPTION
**Pull request by gurukiran07**
Link to current `Series.agg`: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.agg.html

Original question asked in gitter:
Does `pd.Series.agg` with `func` parameter set to `'median'` uses `np.nanmedian`(Not only `median` including `mean`, `mode`)?
```python
s= pd.Series([np.nan, np.nan,1,1,1])
s.agg('median') 
# 1
s.agg(np.median)
# 1
np.median(s.to_numpy())
# nan
np.nanmedian(s.to_numpy())
# 1
```
Whenever `NaN` is present does it fallback to using `np.nanmedian`?

---

Reply from @MarcoGorelli 
if you look at `pandas/core/base.py` you'll see
```
    np.median: "median",
    np.nanmedian: "median",
```
in `_cython_table`. So, both resolve to the same internal cython function.

---

IMO it's better to mention this in the docs under `Note:` section.

Under note section saying:
*Some NumPy functions such as `np.mean`, `np.nanmean`, `np.median` etc. resolve to their corresponding internal cython function.*

#### Output of `python validate_docstrings.py pandas.Series.agg`
<details>

```
################################################################################
######################## Docstring (pandas.Series.agg)  ########################
################################################################################

Aggregate using one or more operations over the specified axis.

.. versionadded:: 0.20.0

Parameters
----------
func : function, str, list or dict
    Function to use for aggregating the data. If a function, must either
    work when passed a Series or when passed to Series.apply.

    Accepted combinations are:

    - function
    - string function name
    - list of functions and/or function names, e.g. ``[np.sum, 'mean']``
    - dict of axis labels -> functions, function names or list of such.
axis : {0 or 'index'}
        Parameter needed for compatibility with DataFrame.
*args
    Positional arguments to pass to `func`.
**kwargs
    Keyword arguments to pass to `func`.

Returns
-------
scalar, Series or DataFrame

    The return can be:

    * scalar : when Series.agg is called with single function
    * Series : when DataFrame.agg is called with a single function
    * DataFrame : when DataFrame.agg is called with several functions

    Return scalar, Series or DataFrame.

See Also
--------
Series.apply : Invoke function on a Series.
Series.transform : Transform function producing a Series with like indexes.

Notes
-----
`agg` is an alias for `aggregate`. Use the alias.
Some NumPy functions such as `np.mean`, `np.nanmean`, `np.median` etc.
resolve to their corresponding internal cython function.

A passed user-defined-function will be passed a Series for evaluation.

Examples
--------
>>> s = pd.Series([1, 2, 3, 4])
>>> s
0    1
1    2
2    3
3    4
dtype: int64

>>> s.agg('min')
1

>>> s.agg(['min', 'max'])
min   1
max   4
dtype: int64

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Series.agg" correct. :)
```
</details>

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5074,7 +5074,9 @@ def pipe(self, func, *args, **kwargs):
         {see_also}
         Notes
         -----
-        `agg` is an alias for `aggregate`. Use the alias.
+        - `agg` is an alias for `aggregate`. Use the alias.
+        - Some NumPy functions such as ``np.mean``, ``np.nanmean``, ``np.median`` etc.
+            resolve to their corresponding internal cython function.
```
Comment (MarcoGorelli) [inline pandas/core/generic.py:5079]: Thanks @gurukiran07 - is the second line more intented than it needs to be?
```suggestion
        - Some NumPy functions such as ``np.mean``, ``np.nanmean``, ``np.median`` etc.
          resolve to their corresponding internal Cython function.
```

Second, I wonder if the significance of this is clear to a user - would it be clearer with an example in the "Examples" section? Perhaps show that you get the same output from `.agg(np.nanmedian)` and `.agg(np.median)`?

Comment (gurukiran07) [reply pandas/core/generic.py:5079]: Yes, agreed but it would be little distracting to add implementation details in the example IMO. Would be better if we elaborate a little more in the notes by adding *`.agg(np.median)`, `.agg(np.nanmedian)` will be handled in the same by the corresponding Cython function and similarly for `np.mean`, `np.nanmean` etc* What do you think?

Comment (MarcoGorelli) [reply pandas/core/generic.py:5079]: Sure - or just mention that, in general, pandas operations exclude NaNs

Comment (gurukiran07) [reply pandas/core/generic.py:5079]: @MarcoGorelli Perfect. Will edit it and push it. Thank you. And one more thing this applies to `transform` also right since `transform` is `agg` under the hood right? Should I add it there too?

Comment (gurukiran07) [reply pandas/core/generic.py:5079]: Should I mention *pandas operations like `"mean"`, `"median"` etc.* to make it more obvious. what do you think?

Comment (MarcoGorelli) [reply pandas/core/generic.py:5079]: How about (feel free to reword, of course)
```suggestion
        - Some NumPy functions resolve to their corresponding internal Cython function. As pandas operations
          generally exclude NaNs, this means that `.agg(np.nanmedian)`, `.agg(np.median)`, and `.agg('median')
          will return the same result.
```
?

I'll check about transform later (but if you've checked the source code already and have seen that that's the case, feel free to add this sentence to the transform docstring too)

Comment (gurukiran07) [reply pandas/core/generic.py:5079]: > How about  (feel free to reword, of course)

This is perfectly on point and short. I'll add these to transform too.

Link to transform  https://github.com/pandas-dev/pandas/blob/v1.0.5/pandas/core/generic.py#L10423
```python
@Appender(_shared_docs["transform"] % dict(axis="", **_shared_doc_kwargs))
    def transform(self, func, *args, **kwargs):
        result = self.agg(func, *args, **kwargs)
        if is_scalar(result) or len(result) != len(self):
            raise ValueError("transforms cannot produce aggregated results")

        return result
```

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5075,9 +5075,9 @@ def pipe(self, func, *args, **kwargs):
         Notes
         -----
         - `agg` is an alias for `aggregate`. Use the alias.
-        - Some NumPy functions resolve to their corresponding internal Cython function. As pandas operations
-          generally exclude NaNs, this means that `.agg(np.nanmedian)`, `.agg(np.median)`, and `.agg('median')
-          will return the same result.
+        - Some NumPy functions resolve to their corresponding internal Cython function.
+          As pandas operations generally exclude NaNs, for example `.agg(np.nanmedian)`,
+          `.agg(np.median)`, and `.agg('median') will return the same result.
```
Comment (MarcoGorelli) [inline pandas/core/generic.py:5080]: Cool, IMO this documents the behaviour nicely without being too long. Just a suggestion on the wording (probably needs reformatting) 
```
For example, `.agg(np.nanmedian)`, `.agg(np.median)`, and `.agg('median') all return the same result (note that pandas operations generally exclude NaNs).
```
if after this you ping a core member (e.g. datapythonista) I'd like to think it'd get merged relatively quickly (but there's _lots_ of open issues, so please be patient if it takes some time)

Comment (gurukiran07) [reply pandas/core/generic.py:5080]: @MarcoGorelli Yes, perfect. Will update with changes. Should I update `transform` docstring too? I'll keep the same wording but replace `.agg` with `.transform`. 

Yes, will ping him once PR passes all the checks. Thank you.

**Comment by gurukiran07** [comment]
@MarcoGorelli I guess this is fixed now. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.agg.html

**Comment by MarcoGorelli** [comment]
@gurukiran07 why did you close this, and why do you say "I guess this is fixed now"?

**Comment by gurukiran07** [comment]
@MarcoGorelli 
```
The aggregation operations are always performed over an axis, either the
index (default) or the column axis. This behavior is different from
numpy aggregation functions (mean, median, prod, sum, std,
var), where the default is to compute the aggregation of the flattened
array, e.g., numpy.mean(arr_2d) as opposed to
numpy.mean(arr_2d, axis=0).
```
This was mentioned in the docs. I thought this would cover `NaNs` part, but this doesn't look like it. Reopening the PR.

**Comment by gurukiran07** [comment]
@datapythonista  @MarcoGorelli It's green now, all tests passed. If there are any changes to be made let me know.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5076,6 +5076,9 @@ def pipe(self, func, *args, **kwargs):
         -----
         `agg` is an alias for `aggregate`. Use the alias.
 
+        Pandas operations generally exclude NaNs. For example, `agg(np.nanmedian)`,
+        `agg(np.median)`, and `agg('median')` will return the same result.
```
Comment (MarcoGorelli) [inline pandas/core/generic.py:5080]: I think you'll need double backticks here to get this to display as code, e.g.
```
``agg(np.median)``
```
instead of 
```
`agg(np.median)`
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Minor comment, other than that looks good to me

File: pandas/core/generic.py (hunk 3 above)
Comment (gurukiran07) [reply pandas/core/generic.py]: Yes sure, fixed that. I was hesitant to use ` `` ` because I thought during docstring validation anything b/w ` `` ` would executed.

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5076,6 +5076,9 @@ def pipe(self, func, *args, **kwargs):
         -----
         `agg` is an alias for `aggregate`. Use the alias.
 
+        Pandas operations generally exclude NaNs. For example, ``agg(np.nanmedian)``,
+        ``agg(np.median)``, and ``agg('median')`` will return the same result.
```
Comment (datapythonista) [inline pandas/core/generic.py:5080]: Thanks @gurukiran07, but I don't find this comment very clear. I don't think we want to talk about what pandas operations generally do in this docstring. I guess you mean that `aggregate` ignores `NaN` values, even if the function argument does not. Just say that, with a code example if you want to be clearer. But I don't expect readers of this docstring to know about `numpy.nanmedian` and `numpy.median` to use them as an example, after a vague sentence about pandas operations in general.

Can you rephrase please?

**Review by datapythonista — APPROVED** [review]
Thanks @gurukiran07, looks good to me.

**Comment by gurukiran07** [comment]
> Thanks @gurukiran07, looks good to me.

@datapythonista Guess, I need to retrigger the checks. Will ping once green.

**Comment by gurukiran07** [comment]
@datapythonista All checks passed, it's green now.

**Comment by MarcoGorelli** [comment]
Cool, merging then following Marc's approval, thanks @gurukiran07
